lua/gitblame/init.lua

```diff
@@ -627,12 +619,84 @@ local function clear_all_extmarks()
     end
 end
 
+-- debounce function
+--- @param func function the function which will be wrapped
+--- @param delay  integer time, millisecond
+local debounce = function(func, delay)
+    local timer = nil
+    return function(...)
+        local args = { ... }
+        if timer then
+            timer:stop()
+            timer = nil
+        end
+
+        timer = vim.defer_fn(function()
+            func(unpack(args))
+            timer = nil
+        end, delay)
+    end
+end
+
+-- this function is uesed to verify the config info for debounce
+--- @return boolean
+local verify_debounce_info = function()
```

Comment:
Rather than implementing a hard fail if the values don't match the expected values, it might be better to add this in another pull request since we don't really do any value checking on any of the configuration values and I think it would be good to separately add it for all configuration values. Thoughts?

Reply:
We could do something like `validate_configuration_options` which validates all the values at once

Reply:
I don't know all the configurations, so this is all I can do here

Reply:
Ok, I can look into adding something like this after this pull request